exchange/exchange.go

```diff
@@ -284,14 +290,18 @@ func updateHbPbCatDur(bid *pbsOrtbBid, dealTierInfo *DealTierInfo, bidCategory m
 	}
 }
 
-func (e *exchange) makeAuctionContext(ctx context.Context, needsCache bool) (auctionCtx context.Context, cancel context.CancelFunc) {
+func (e *exchange) makeAuctionContext(ctx context.Context, needsCache bool, debugInfo bool) (auctionCtx context.Context, cancel context.CancelFunc) {
 	auctionCtx = ctx
 	cancel = func() {}
 	if needsCache {
 		if deadline, ok := ctx.Deadline(); ok {
 			auctionCtx, cancel = context.WithDeadline(ctx, deadline.Add(-e.cacheTime))
 		}
 	}
+	// Don't populate context unless `debugInfo` is true
+	if debugInfo {
+		auctionCtx = context.WithValue(auctionCtx, DebugContextKey("debugInfo"), "true")
```

Comment:
Why populate it only when it's true? We're anyways gonna try and retrieve it and rather than checking for a nil it makes more sense to check for a `true`/`false` value IMHO

Reply:
Given that the `Context` library offers a capability to store and retrieve values, kind of like a hash table or hash map, why do we need to store an extra value there when `debugInfo` is `false`? Why would we need that extra info?

Reply:
Modified to boolean value as requested.
```
auctionCtx = context.WithValue(auctionCtx, DebugContextKey("debugInfo"), debugInfo)
```